src/Microsoft.ML.Data/Model/Onnx/SaveOnnxCommand.cs

```diff
@@ -228,10 +266,68 @@ private void Run(IChannel ch)
 
             if (!string.IsNullOrWhiteSpace(Args.OutputModelFile))
             {
+                Contracts.Assert(loader != null);
+
                 ch.Trace("Saving the data pipe");
                 // Should probably include "end"?
                 SaveLoader(loader, Args.OutputModelFile);
             }
         }
+
+        public sealed class Output
+        {
+            //REVIEW: Would be nice to include ONNX protobuf model here but code generator needs an upgrade.
+        }
+
+        //REVIEW: Ideally there is no need to define this input class and just reuse the Argument class from SaveONNX command
+        //but the code generator cannot parse certain complicated data types in the base class that Argument class extends.
+        //We should fix the code generator and use the Argument class.
```

Comment:
OK. What types specifically? Are we certain that these aren't types that should be set to command line only?

Reply:
these are cmd line only types:
            [Argument(ArgumentType.Multiple, HelpText = "Transform", ShortName = "xf")]
            public KeyValuePair<string, SubComponent<IDataTransform, SignatureDataTransform>>[] Transform;

---
In reply to: [191822211](https://github.com/dotnet/machinelearning/pull/248#discussion_r191822211) [](ancestors = 191822211)

Reply:
OK cool. Set those visible to command line. These shouldn't be going into a command line *anyway*. You are given the datamodel in entry-point land (since the point is you're building them up programmatically anyway) but command-line land has to specify these things.

---
In reply to: [191901676](https://github.com/dotnet/machinelearning/pull/248#discussion_r191901676) [](ancestors = 191901676,191822211)